cpp/src/parquet/page_index.cc

```diff
@@ -441,8 +442,11 @@ RowGroupIndexReadRange PageIndexReader::DeterminePageIndexRangesInRowGroup(
         throw ParquetException("Invalid index location: offset ", index_location->offset,
                                " length ", index_location->length);
       }
+      int64_t index_end = 0;
+      ::arrow::internal::AddWithOverflow(index_location->offset, index_location->length,
```

Comment:
Hmm, ok, but you should check the return value, which is the whole point.

Reply:
Filed https://github.com/apache/arrow/issues/34003 to make this mandatory.

Reply:
> Hmm, ok, but you should check the return value, which is the whole point.

My bad! Sorry about that